pybossa/sched.py

```diff
@@ -590,20 +591,28 @@ def acquire_lock(task_id, user_id, limit, timeout, pipeline=None, execute=True):
     return False
 
 
-def release_reserve_task_lock_by_id(project_id, task_id, user_id, timeout, expiry=EXPIRE_RESERVE_TASK_LOCK_DELAY):
+def release_reserve_task_lock_by_id(project_id, task_id, user_id, timeout, expiry=EXPIRE_RESERVE_TASK_LOCK_DELAY, release_all_task=None):
```

Comment:
is `release_all_tasks` parameter necessary? Would it be more intuitive to pass `task_id=None` to indicate that we want to release all task_lock_category?

Reply:
```release_all_tasks``` is necessary because we cannot use "task_id" to distinguish the cases.

Reply:
makes sense, thanks.

Another minor comment, `release_all_task` is a Boolean variable, `None` is ok but would it be better to use `False` as default value?

Reply:
Makes sense. Will default it to "False". Thank you.